### PR TITLE
Fix path processing

### LINKED ISSUE #2856: FATAL_ERROR when directory name used as input filename
ISPC fails with FATAL_ERROR for the following command:

```bash
$ ispc -g ../
util.cpp:535: Assertion failed: "basenameStart[0] != '\0'".
***
*** Please file a bug report at https://github.com/ispc/ispc/issues
*** (Including as much information as you can about how to reproduce this error).
*** You have apparently encountered a bug in the compiler that we'd like to fix!
***
/localdisk2/anurmukh/ispc/src/main.cpp(406): FATAL ERROR: Unhandled signal sent to process; terminating.
Aborted
```

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR fixes #2856

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1100,6 +1100,20 @@ int main(int Argc, char *Argv[]) {
         exit(1);
     }
 
+    if (strcmp(file, "-")) {
```
Comment (dbabokin) [inline src/main.cpp:1103]: `strncmp`, otherwise it will not pass security review.

File: src/util.cpp [hunk 2]
```diff
@@ -503,39 +505,23 @@ std::vector<std::string> ispc::MatchStrings(const std::string &str, const std::v
...
+    if (llvm::sys::path::is_absolute(relativeName)) {
+        // We may actually get not relative path but absolute, then just use it.
         fullPath = relativeName;
-    else {
-        fullPath = g->currentDirectory;
-        if (fullPath[fullPath.size() - 1] != '/')
-            fullPath.push_back('/');
-        fullPath += relativeName;
+    } else {
+        fullPath = currentDirectory;
```
Comment (dbabokin) [inline src/util.cpp:516]: Interesting, it used `g->currentDirectory` before. Is it something different?

Comment (nurmukhametov) [reply src/util.cpp:516]: Both use cases have `currentDirector == g->currentDirectory` so nothing changed except the removed inconsistency.

File: src/main.cpp (hunk 1 above)
Comment (nurmukhametov) [reply src/main.cpp]: fixed

**Review by dbabokin — APPROVED** [review]
LGTM

**Pull request merged**